### PR TITLE
Allow smoke tests to be run against docker stack

### PR DESCRIPTION
**Pull request by ollyjshaw**

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -397,3 +400,6 @@ DEPENDENCIES
   therubyracer
   uglifier (~> 2.0.1)
   wraith (~> 1.3.0)
+
+BUNDLED WITH
+   1.10.6
```
Comment (paulrobinson) [inline Gemfile.lock]: I think you need to remove this change, right?

**Comment by paulrobinson** [comment]
@LightGuard can you review this please?

File: .awestruct_ignore [hunk 2]
```diff
@@ -13,3 +13,6 @@ add-your-event.html.slim
 stylesheets/fonts/SIL Open Font License.txt
 stylesheets/rh-icons
 CONTRIBUTING.md
+_docker
+_cucumber
+cucumber_failures
```
Comment (LightGuard) [inline .awestruct_ignore]: `cucumber_failures` needs to be added, but `_docker` and `_cucumber` don't need to be. They start with `_` which is auto-ignored by awestruct.

File: Gemfile.lock (hunk 1 above)
Comment (LightGuard) [inline Gemfile.lock]: Why would this need to be removed?

File: README.md [hunk 3]
```diff
@@ -140,6 +140,10 @@ To run the acceptance tests (where HOST_TO_TEST is the host you are interested i
 
 `bundle exec ./control.rb ---acceptance_test_target[=HOST_TO_TEST]`
 
+To run the acceptance tests against the locally running docker stack:
+
+`bundle exec ./control.rb ---acceptance_test_docker`
```
Comment (LightGuard) [inline README.md]: I think you have an extra `-` in there, should only be two, not three, same with line 141.

File: Rakefile [hunk 4]
```diff
@@ -106,7 +106,9 @@ end
 
 desc 'Build and preview the site locally in development mode'
 task :preview, [:profile] => :check do |task, args|
-  run_awestruct "-P #{args[:profile] || 'development'} -a -s --force -q --auto --livereload -b 0.0.0.0"
+  profile = args[:profile] || 'development'
+  awestruct_running_port = awestruct_port(profile)
+  run_awestruct "-P #{profile} -a -s --force -q --auto --livereload -b 0.0.0.0 -p #{awestruct_running_port}"
```
Comment (LightGuard) [inline Rakefile]: We should change `-b 0.0.0.0` to `-b localhost`

File: Rakefile [hunk 5]
```diff
@@ -421,6 +423,15 @@ def which(cmd, opts = {})
   return $awestruct_cmd
 end
 
+def awestruct_port(profile)
+  if profile.to_sym==:docker
```
Comment (LightGuard) [inline Rakefile]: Please follow the whitespace around operations like the rest of the file

Comment (LightGuard) [inline Rakefile]: Also what happens if we use drupal? That should be added as well.

File: _cucumber/features/support/env.rb [hunk 6]
```diff
@@ -18,7 +18,13 @@
 SCREENSHOT_DIRECTORY = '_cucumber/screenshots'
 
 Capybara.configure do |config|
-  host_to_test = ENV['HOST_TO_TEST'] ||= 'http://docker:32768'
+
+  if ENV['HOST_TO_TEST']== ''
```
Comment (LightGuard) [inline _cucumber/features/support/env.rb]: `== ` => ` == ` please

File: Gemfile.lock (hunk 1 above)
Comment (ollyjshaw) [inline Gemfile.lock]: As long as the message is in full, this is fine.

File: Rakefile (hunk 4 above)
Comment (ollyjshaw) [inline Rakefile]: This is not a change I am bringing in. Ok if we look at this in a separate change outside of this PR?

**Comment by ollyjshaw** [comment]
retest this please

File: Rakefile (hunk 4 above)
Comment (LightGuard) [inline Rakefile]: Sure, no problem

Comment (ollyjshaw) [inline Rakefile]: Thanks, I'll make a note to do it now.

**Comment by ollyjshaw** [comment]
retest this please

**Pull request merged**